homeassistant/components/sensor/knx.py

```diff
@@ -0,0 +1,124 @@
+#  Temperature as defined in KNX Standard 3.10 - 9.001 DPT_Value_Temp
+KNX_TEMP_MIN = -273
+KNX_TEMP_MAX = 670760
+
+#  Luminance(LUX) as Defined in KNX Standard 3.10 - 9.004 DPT_Value_Lux
+KNX_LUX_MIN = 0
+KNX_LUX_MAX = 670760
+
+#  Speed m/s as defined in KNX Standard 3.10 - 9.005 DPT_Value_Wsp
+KNX_SPEED_MS_MIN = 0
+KNX_SPEED_MS_MAX = 670760
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Setup the KNX Sensor platform."""
+    # Add KNX Temperature Sensors
+    # KNX Datapoint 9.001 DPT_Value_Temp
+    if config["type"] == "temperature":
+        minimum_value, maximum_value = \
+            update_and_define_min_max(config, KNX_TEMP_MIN,
+                                      KNX_TEMP_MAX)
+
+        add_entities([
+            KNXSensorFloatClass(hass, KNXConfig(config), TEMP_CELSIUS,
+                                minimum_value, maximum_value)
+        ])
+
+    # Add KNX Speed Sensors(Like Wind Speed)
+    # KNX Datapoint 9.005 DPT_Value_Wsp
+    if config["type"] == "speed_ms":
+        minimum_value, maximum_value = \
+            update_and_define_min_max(config, KNX_SPEED_MS_MIN,
+                                      KNX_SPEED_MS_MAX)
+
+        add_entities([
+            KNXSensorFloatClass(hass, KNXConfig(config), SPEED_METERPERSECOND,
+                                minimum_value, maximum_value)
+        ])
+
+    # Add KNX Illuminance Sensors(Lux)
+    # KNX Datapoint 9.004 DPT_Value_Lux
+    if config["type"] == "illuminance":
+        minimum_value, maximum_value = \
+            update_and_define_min_max(config, KNX_LUX_MIN, KNX_LUX_MAX)
+
+        add_entities([
+            KNXSensorFloatClass(hass, KNXConfig(config), ILLUMINANCE_LUX,
+                                minimum_value, maximum_value)
+        ])
+
+
+def update_and_define_min_max(config, minimum_default,
+                              maximum_default):
+    """Function help determinate a min/max value defined in config."""
+    minimum_value = minimum_default
+    maximum_value = maximum_default
+    if config.get('minimum'):
+        minimum_value = config.get('minimum')
+
+    if config.get('maximum'):
+        maximum_value = config.get('maximum')
+
+    return minimum_value, maximum_value
+
+
+class KNXSensorBaseClass():  # pylint: disable=too-few-public-methods
+    """Sensor Base Class for all KNX Sensors."""
+
+    @property
+    def cache(self):
+        """We don't want to cache any Sensor Value."""
+        return False
+
+
+class KNXSensorFloatClass(KNXGroupAddress, KNXSensorBaseClass):
+    """
+    Base Implementation of a 2byte Floating Point KNX Telegram.
+
+    Defined in KNX 3.7.2 - 3.10
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, config, unit_of_measurement, minimum_sensor_value,
+                 maximum_sensor_value):
+        """Initialize a KNX Float Sensor."""
+        self._unit_of_measurement = unit_of_measurement
```

Comment:
Dont want to but it in there, there are i think KNX Datatypes without a Max Min Value so they would extend the KNXSensorBaseClass and got other Init Parameters